datafusion/core/tests/sql/aggregates.rs

```diff
@@ -99,11 +99,11 @@ async fn aggregate_timestamps_count() -> Result<()> {
     .await;
 
     let expected = vec![
-        "+--------------+---------------+---------------+-------------+",
-        "| COUNT(nanos) | COUNT(micros) | COUNT(millis) | COUNT(secs) |",
-        "+--------------+---------------+---------------+-------------+",
-        "| 3            | 3             | 3             | 3           |",
-        "+--------------+---------------+---------------+-------------+",
+        "+----------------+-----------------+-----------------+---------------+",
+        "| COUNT(t.nanos) | COUNT(t.micros) | COUNT(t.millis) | COUNT(t.secs) |",
+        "+----------------+-----------------+-----------------+---------------+",
+        "| 3              | 3               | 3               | 3             |",
+        "+----------------+-----------------+-----------------+---------------+",
```

Comment:
i feel this might be an unwanted change, since it doesn't seem intuitive to do a `select count(nanos)` and have the result be `COUNT(t.nanos)` in my opinion 🤔

especially as i believe this might be only for counts which utilize aggregate statistics rule? so regular counts would still output `COUNT(nanos)` i believe?

Reply:
yes,i agree with you ,  regular counts output COUNT(nanos) will feel more intuitive.

but in the current situation, we need to ensure the output remains consistent, regardless of whether the AggregateStatistics optimizer is used or not.

at main branch, disable aggregate_statistics or enable will retrun difference resutl.
```
#[tokio::test]
async fn count_test() -> Result<()> {
    let mut ctx = SessionContext::new();
    let disable_aggregate_statistics = true;

    //disable or not,this test case should be passed
    if (disable_aggregate_statistics) {
        let with_out_aggregate_statistics = ctx
            .state()
            .physical_optimizers()
            .iter()
            .filter(|optimizer| optimizer.as_ref().name().ne("aggregate_statistics"))
            .map(|optimizer| optimizer.clone())
            .collect();
        let state = ctx
            .state()
            .with_physical_optimizer_rules(with_out_aggregate_statistics);
        ctx = SessionContext::with_state(state);
    }

    let testdata = datafusion::test_util::parquet_test_data();
    ctx.register_parquet(
        "alltypes_tiny_pages",
        &format!("{testdata}/alltypes_tiny_pages.parquet"),
        ParquetReadOptions::default(),
    )
    .await?;
    let results = ctx
        .sql("SELECT count(id), max(id), min(id) FROM alltypes_tiny_pages")
        .await?
        .collect()
        .await?;
    let expected = vec![
        "+-------------------------------+-----------------------------+-----------------------------+",
        "| COUNT(alltypes_tiny_pages.id) | MAX(alltypes_tiny_pages.id) | MIN(alltypes_tiny_pages.id) |",
        "+-------------------------------+-----------------------------+-----------------------------+",
        "| 7300                          | 7299                        | 0                           |",
        "+-------------------------------+-----------------------------+-----------------------------+",
    ];
    assert_batches_sorted_eq!(expected, &results);

    Ok(())
}

```

Reply:
Ok, if they're consistent then that's good

Reply:
I double checked that the query plan shows `t.col` if there are no statistics by using datafusion-cli not from this branch 👍 

```sql
❯ select count(a) from './datafusion/core/tests/data/example.csv' as t limit 1;
+------------+
| COUNT(t.a) |
+------------+
| 1          |
+------------+
```